cargo-dist/src/build/cargo.rs

```diff
@@ -201,11 +201,26 @@ pub fn build_cargo_target(
         };
         match message {
             cargo_metadata::Message::CompilerArtifact(artifact) => {
-                let Some(new_exe) = artifact.executable else {
-                    continue;
+                if let Some(new_exe) = artifact.executable {
+                    // Hey we got an executable, record that fact
+                    expected.found_bin(
+                        artifact.package_id.to_string(),
+                        new_exe,
+                        artifact.filenames,
+                    );
+                } else {
+                    // "executable" is singled out as an artifact
+                    // type in cargo's output; for libraries, we have
+                    // to just look at the list of output files
+                    // and let found_bin sort it out.
+                    for new_exe in &artifact.filenames {
+                        expected.found_bin(
+                            artifact.package_id.to_string(),
+                            new_exe.to_owned(),
+                            artifact.filenames.clone(),
```

Comment:
Yeah, it may make more sense for a `found_bins` that takes multiple paths. Really it's just hedging on the fact here that the full list of filenames is passed for the random other stuff that might be there (symbols, etc.), but the single-filename field is where we put maybe-requested files.